core/src/main/java/software/amazon/awssdk/core/internal/http/response/AwsErrorResponseHandler.java

```diff
@@ -55,16 +56,16 @@ private SdkBaseException handleAse(HttpResponse response, ExecutionAttributes ex
         } catch (Exception e) {
             // If the errorResponseHandler doesn't work, then check for error responses that don't have any content
             if (statusCode == HttpStatusCodes.REQUEST_TOO_LONG) {
-                AmazonServiceException exception = new AmazonServiceException("Request entity too large");
-                exception.setStatusCode(statusCode);
-                exception.setErrorType(AmazonServiceException.ErrorType.Client);
-                exception.setErrorCode("Request entity too large");
+                SdkServiceException exception = new SdkServiceException("Request entity too large");
```

Comment:
should this be a `SdkClientException` since it's a client error? would it retry if we use `SdkServiceException` here?

Reply:
Not really. We still pull out the status code and status text associated with the response from the service. The SdkClientException doesn't have these fields. It also doesn't portray as well that the request was made and the service rejected it as the clients fault.

Reply:
I'm actually thinking this might be dead code. The error response handler should always be able to throw an unmarshalled exception no matter what happens.